lib/ta_wifi/ta_wifi.c

```diff
@@ -0,0 +1,1100 @@
+    te_errno rc;
+
+    UNUSED(gid);
+    UNUSED(oid);
+    UNUSED(empty);
+
+    ENTRY("%s", port_name);
+
+    port = ta_wifi_find_port(port_name);
+    if (port == NULL)
+        return TE_RC(TE_TA_UNIX, TE_ENOENT);
+
+    rc = te_strtou_size(value, 0, &port->channel,
+                        sizeof(port->channel));
+    return TE_RC_UPSTREAM(TE_TA_UNIX, rc);
+}
+
+static te_errno
+node_wifi_configurator_get(unsigned int gid, const char *oid, char *value,
+    const char *empty)
+{
+    te_errno rc;
+
+    UNUSED(gid);
+    UNUSED(oid);
+    UNUSED(empty);
+
+    rc = te_snprintf(value, RCF_MAX_VAL, "%s",
+             te_enum_map_from_value(wifi_configurator_mapping,
+                                    ta_wifi_get_node()->configurator));
+    return TE_RC_UPSTREAM(TE_TA_UNIX, rc);
+}
+
+static te_errno
+node_wifi_configurator_set(unsigned int gid, const char *oid,
+    const char *value, const char *empty)
+{
+    int mapped;
+
+    UNUSED(gid);
+    UNUSED(oid);
+    UNUSED(empty);
+
+    mapped = te_enum_map_from_str(wifi_configurator_mapping, value, -1);
+    if (mapped < 0)
+        return TE_RC(TE_TA_UNIX, TE_EINVAL);
+
+    ta_wifi_get_node()->configurator = mapped;
+    return 0;
+}
+
+static te_errno
+node_wifi_port_enable_get(unsigned int gid, const char *oid, char *value,
+    const char *empty, const char *port_name)
+{
+    ta_wifi_port *port;
+
+    UNUSED(gid);
+    UNUSED(oid);
+    UNUSED(empty);
+
+    port = ta_wifi_find_port(port_name);
+    if (port == NULL)
+        return TE_RC(TE_TA_UNIX, TE_ENOENT);
+
+    sprintf(value, "%d", port->enable ? 1 : 0);
+
+    return 0;
+}
+
+static te_errno
+node_wifi_port_enable_set(unsigned int gid, const char *oid,
+    const char *value, const char *empty, const char *port_name)
+{
+    ta_wifi_port *port;
+
+    UNUSED(gid);
+    UNUSED(oid);
+    UNUSED(empty);
+
+    port = ta_wifi_find_port(port_name);
+    if (port == NULL)
+        return TE_RC(TE_TA_UNIX, TE_ENOENT);
+
+    port->enable = (atoi(value) > 0) ? true : false;
+
+    return ta_unix_conf_wifi_apply();
+}
+
+static te_errno
+node_wifi_port_ifname_get(unsigned int gid, const char *oid, char *value,
+    const char *empty, const char *port_name)
+{
+    ta_wifi_port *port;
+
+    UNUSED(gid);
+    UNUSED(oid);
+    UNUSED(empty);
+
+    port = ta_wifi_find_port(port_name);
+    if (port == NULL)
+        return TE_RC(TE_TA_UNIX, TE_ENOENT);
+
+    if (port->ifname != NULL)
+        strcpy(value, port->ifname);
+    else
+        strcpy(value, "");
+
+    return 0;
+}
+
+static te_errno
+node_wifi_port_ifname_set(unsigned int gid, const char *oid,
+    const char *value, const char *empty, const char *port_name)
+{
+    ta_wifi_port *port;
+
+    UNUSED(gid);
+    UNUSED(oid);
+    UNUSED(empty);
+
+    port = ta_wifi_find_port(port_name);
+    if (port == NULL)
+        return TE_RC(TE_TA_UNIX, TE_ENOENT);
+
+    port->ifname = strdup(value);
+    if (port->ifname == NULL)
+        return TE_RC(TE_TA_UNIX, TE_ENOMEM);
```

Comment:
`TE_STRDUP` without a null check, please.

Reply:
You're reviewing old version.